realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyMediatorGenerator.java

```diff
@@ -177,7 +180,7 @@ private void emitGetSimpleClassNameMethod(JavaWriter writer) throws IOException
         emitMediatorShortCircuitSwitch(new ProxySwitchStatement() {
             @Override
             public void emitStatement(int i, JavaWriter writer) throws IOException {
-                writer.emitStatement("return %s.getSimpleClassName()", qualifiedProxyClasses.get(i));
+                writer.emitStatement("return \"%s\"", internalClassNames.get(i));
```

Comment:
That does seem more Proguard-friendly.

Reply:
Not to mention correct 😉 The previous implementation did not work if you renamed the class name using `@RealmClass(name = "othername")`.